### PR TITLE
Apply nullability attributes to TimeChangedEventArgs

### LINKED ISSUE #29513: Apply nullability attributes to TimeChangedEventArgs type as well
### Description

Nullability changes for DatePicker and TimePicker.

It seems nullability not applied to TimeChangedEventArgs type.

https://github.com/dotnet/maui/blob/84331b3c4afccd2b3524c96a10500a9c2a296280/src/Controls/src/Core/TimeChangedEventArgs.cs#L20-L28

### Public API Changes

Type of both the `NewTime` and `OldTime` properties is to be updated as `TimeSpan?`

### Intended Use-Case

Nullability changes

### PR DESCRIPTION
**Pull request by jsuarezruiz**
### Description of Change

In .NET 10 we make the Time property of the TimePicker class nullable https://github.com/dotnet/maui/pull/27930 However missed to apply nullability attributes to **TimeChangedEventArgs**. With the changes from this PR we align the behavior with the DatePicker https://github.com/dotnet/maui/blob/net10.0/src/Controls/src/Core/DateChangedEventArgs.cs

### Issues Fixed

Fixes #29513 
Related with https://github.com/dotnet/maui/pull/27930

### CODE REVIEW

File: src/Controls/tests/Core.UnitTests/TimePickerUnitTest.cs [hunk 1]
```diff
@@ -88,8 +88,8 @@ public void DatePickerSelectedEventArgs(TimeSpan initialTime, TimeSpan finalTime
 			timePicker.Time = initialTime;
 
 			TimePicker pickerFromSender = null;
-			TimeSpan oldTime = new TimeSpan();
-			TimeSpan newTime = new TimeSpan();
+			TimeSpan? oldTime = new TimeSpan();
```
[the remark this answers is not in the document]
Comment (jsuarezruiz) [reply src/Controls/tests/Core.UnitTests/TimePickerUnitTest.cs:91]: Could be. In this case following the same pattern done with the changes in DatePicker (handler, tests etc): https://github.com/dotnet/maui/blob/net10.0/src/Controls/tests/Core.UnitTests/DatePickerUnitTest.cs#L179

[the remark this answers is not in the document]
Comment (rmarinho) [reply src/Controls/tests/Core.UnitTests/TimePickerUnitTest.cs:91]: Ok, we might want to create a couple of more tests with these variables taking null and everything works too.

**Review by rmarinho — APPROVED** [review]
